tests/ci/test_test_quality_guardrails.py

```
@@ -501,3 +539,164 @@ def test_changed_tests_have_no_sole_isinstance_assertions() -> None:
+    as a rationalized "covers both branches" — but it covers nothing: it
+    passes even when the function raises, returns a truthy-but-wrong value,
+    or leaks state.
+
+    Also flags the symmetric ``X is None or X is not None``.
+    """
+    try:
+        tree = ast.parse(source, filename=path)
+    except SyntaxError:
+        return []
+
+    def _is_is_none(n: ast.AST, negated: bool) -> str | None:
```

Ruff (0.15.10)

[warning] Boolean-typed positional argument in function definition

(FBT001)